src/serialize.h

```diff
@@ -59,6 +59,12 @@ inline T* NCONST_PTR(const T* val)
     return const_cast<T*>(val);
 }
 
+//! Safely convert odd char pointer types to standard ones.
+static inline char* CharCast(char* c) { return c; }
+static inline char* CharCast(unsigned char* c) { return (char*)c; }
+static inline const char* CharCast(const char* c) { return c; }
+static inline const char* CharCast(const unsigned char* c) { return (const char*)c; }
```

Comment:
why not `reinterpret_cast`?

Reply:
For primitive types there is no distinction between c-style-cast, reinterpret_cast, static_cast.